src/nvim/os/env.c

```diff
@@ -76,6 +76,31 @@ char *os_getenvname_at_index(size_t index)
   return name;
 }
 
+/// Check if the environment variable `name` exists and has
+/// a value of either 1, "on", "yes", or "true".
+bool os_env_istruthy(const char *name) FUNC_ATTR_NONNULL_ALL
+{
+  const char *val = os_getenv(name);
+  if (val == NULL || *val == NUL) {  // TODO(Pyrohh): Remove the check for
+    return false;                    // NUL after #2574 is merged.
+  }
+
+  errno = 0;
+  long n = strtol(val, NULL, 10);
+  if (errno == ERANGE || errno == EINVAL) {
```

Comment:
> Is it possible that errno is some other non-zero value?

No, at least according to POSIX; given that, there's no need to reset it on line 94. Besides, any time we need to deal with errno it should be set to 0 before calling whatever function, so I'm pretty sure it's fine.